File: Frontend/src/models/todo.ts
export enum TodoStatus {
    Todo = 0,
    InProgress = 1,
    Done = 2
}

export interface TodoItem {
    id: number;
    title: string;
    status: TodoStatus;
    todoId: number;
}

export interface Todo {
    id: number;
    title: string;
    icon: string;
    date: string;
    items: TodoItem[];
}

File: Backend/src/TodoWithAI.Domain/Entities/Todo.cs
namespace TodoWithAI.Domain.Entities;

public class Todo
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Icon { get; set; } = string.Empty;
    public DateTime Date { get; set; } = DateTime.UtcNow;

    public List<TodoItem> Items { get; set; } = new();
}
